src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,309 @@
+        this.fillAnimation.setShape(this.switchBackground);
+        this.getChildren().addAll(this.switchBackground, this.switchButton);
+        //Listener
+        this.switchStateBooleanProperty.addListener((observable, oldValue, newValue) -> {
+            boolean tmpIsOn = newValue.booleanValue();
+            this.switchAnimation.setToX(tmpIsOn ? (RECTANGLE_WIDTH_VALUE - (2 * CIRCLE_RADIUS_VALUE)) : 0);
+            this.fillAnimation.setFromValue(tmpIsOn ? RECTANGLE_COLOR_OFF : RECTANGLE_COLOR_ON);
+            this.fillAnimation.setToValue(tmpIsOn ? RECTANGLE_COLOR_ON : RECTANGLE_COLOR_OFF);
+            this.switchTransition.play();
+        });
+        //Mouse listener.
+        this.setOnMouseClicked(event -> this.switchStateBooleanProperty.set(!this.switchStateBooleanProperty.get()));
+    }//</editor-fold>
+    //<editor-fold desc="Constructor with parameters" defaultstate="collapsed">
+    /**
+     * Second constructor with parameters to make the toggle switch configurable. This way if it is needed in different
+     * classes, the switch can be customized by calling for this constructor and passing preferred values to
+     * the parameters.
+     * @param anInitialStateOfBooleanProperty initial boolean state of the switch.
+     * @param aRectangleWidth the width of the rectangle that is used as the background of the switch.
+     * @param aRectangleHeight the height of the rectangle that is used as the background of the switch.
+     * @param aRectanglePositionX the position of the rectangle on the GUI.
+     * @param aCircleRadius radius of the button.
+     * @param aCirclePositionX position of the button on the x-axis of the layout of the GUI.
+     * @param aCirclePositionY position of the button on the y-axis of the layout of the GUI.
+     * @param aDropShadowRadius the radius of the shadow around the circle.
+     * @param aDurationOfTransitionInSeconds the duration of the transition in seconds.
+     * @param aRectangleColorOff the color of the background whe the switch is turned off.
+     * @param aRectangleOutline the color of the outline of the background.
+     * @param aCircleColor the color of the button.
+     * @param aCircleOutline the color of the outline of the button.
+     * @param aCircleShadowColor the color of the shadow around the button.
+     * @param aRectangleColorOn the color of the background when the switch is turned on.
+     */
+    public ToggleSwitch(boolean anInitialStateOfBooleanProperty, int aRectangleWidth, int aRectangleHeight, int aRectanglePositionX, int aCircleRadius,
+                        int aCirclePositionX, int aCirclePositionY, int aDropShadowRadius, double aDurationOfTransitionInSeconds, Color aRectangleColorOff,
+                        Color aRectangleOutline, Color aCircleColor, Color aCircleOutline, Color aCircleShadowColor, Color aRectangleColorOn){
+        this.switchStateBooleanProperty = new SimpleBooleanProperty(anInitialStateOfBooleanProperty);
```

Comment:
Check the parameters and throw IllegalArgumentExceptions if the parameters do not meet the restrictions. E.g. width and height cannot be negative. What other restrictions can you think of?